packages/editor/src/components/DataSource/Api.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useMemo } from 'react';
+import {
+  Box,
+  Text,
+  Input,
+  AccordionItem,
+  AccordionButton,
+  AccordionIcon,
+  AccordionPanel,
+} from '@chakra-ui/react';
+import { ComponentSchema } from '@sunmao-ui/core';
+import { DataSourceItem } from './DataSourceItem';
+
+const COLOR_MAP = {
+  GET: 'green',
+  POST: 'orange',
+  PUT: 'yellow',
+  PATCH: 'yellow',
+  DELETE: 'red',
+};
+
+interface Props {
+  apis: ComponentSchema[];
+  active: string;
+  onItemClick: (api: ComponentSchema) => void;
+  onItemRemove: (api: ComponentSchema) => void;
+}
+
+export const Api: React.FC<Props> = props => {
+  const [search, setSearch] = useState('');
+  const { apis, active, onItemClick, onItemRemove } = props;
+  const list = useMemo(
+    () => apis.filter(({ id }) => id.includes(search)),
+    [search, apis]
+  );
+  const ApiItems = () => (
+    <>
+      {list.map(api => {
+        const trait = api.traits.find(({ type }) => type === 'core/v1/fetch');
+        const properties = trait!.properties;
+        const method = (
+          properties.method as string
+        ).toUpperCase() as keyof typeof COLOR_MAP;
+
+        return (
+          <DataSourceItem
+            key={api.id}
+            dataSource={api}
+            tag={method}
+            name={api.id}
+            active={active === api.id}
+            colorMap={COLOR_MAP}
```

Comment:
`COLOR_MAP` is constant, just import it. No need to pass it as props.

Reply:
`DataSourceItem` component would be used by different components which may not use the same colors, so I pass it as props to make it to be dynamic.